code/game/atoms_health.dm

```diff
@@ -0,0 +1,293 @@
+/atom
+	/// Current health for health processing. Use `get_current_health()`, `damage_health()`, or `restore_health()` for general health references.
+	var/health_current
+	/// Maximum health for simple health processing. Use `get_max_health()` or `set_max_health()` to reference/modify.
+	var/health_max
+
+	/**
+	 * LAZY List of damage type resistance or weakness multipliers, decimal form. Only applied to health reduction. Use `set_damage_resistance()`, `remove_damage_resistance()`, and `get_damage_resistance()` to reference/modify.
+	 *
+	 * Index should be one of the `DAMAGE_` flags.
+	 * Value should be a multiplier that is applied against damage. Values below 1 are a resistance, above 1 are a weakness.
+	 * Value of `0` is considered immunity.
+	 */
+	var/list/health_resistances
+
+	/// Minimum damage required to actually affect health in `can_damage_health()`.
+	var/health_min_damage = 0
+
+/**
+ * Whether or not the atom has health. Checks for a value in `health_max`.
+ * Can be overridden for additional checks.
+ * Returns `TRUE` or `FALSE`.
+ */
+/atom/proc/has_health()
+	SHOULD_CALL_PARENT(TRUE)
+	return !!health_max
+
+/**
+ * Retrieves the atom's current health, or `null` if not using health
+ */
+/atom/proc/get_current_health()
+	SHOULD_CALL_PARENT(TRUE)
+	if (!has_health())
+		return
+	return health_current
+
+/**
+ * Retrieves the atom's maximum health, or `null` if not using health.
+ */
+/atom/proc/get_max_health()
+	SHOULD_CALL_PARENT(TRUE)
+	if (!has_health())
+		return
+	return health_max
```

Comment:
Considering that `health_max` is falsey if the mob has no health this proc chain seems a bit redundant - couldn't you just return max health and let the caller handle it being zero?

Reply:
That's a fair point - Will update